rasa/shared/core/domain.py

```diff
@@ -317,7 +317,11 @@ def merge_domain_dicts(
         if override or domain2.get("session_config"):
             combined[SESSION_CONFIG_KEY] = domain_dict[SESSION_CONFIG_KEY]
 
-        if combined[KEY_INTENTS] or domain_dict[KEY_INTENTS]:
+        if combined.get(KEY_INTENTS) or domain_dict.get(KEY_INTENTS):
+            if not combined.get(KEY_INTENTS):
+                combined[KEY_INTENTS] = []
```

Comment:
```suggestion
            combined[KEY_INTENTS] = combined.get(KEY_INTENTS, [])
```

Reply:
thanks, I knew there was a more elegant way of writing this...